src/data/mv_polynomial/basic.lean

```diff
@@ -541,7 +437,141 @@ lemma as_sum (p : mv_polynomial σ R) : p = ∑ v in p.support, monomial v (coef
 end as_sum
 
 
+
+end semiring
+
+section comm_semiring
+variables [comm_semiring R] {p q : mv_polynomial σ R}
+
+section instances
+
+instance : comm_semiring (mv_polynomial σ R) := add_monoid_algebra.comm_semiring
+instance [semiring R] [semiring S₁] [comm_semiring S₂]
+  [has_scalar R S₁] [semimodule R S₂] [semimodule S₁ S₂] [is_scalar_tower R S₁ S₂] :
+  is_scalar_tower R S₁ (mv_polynomial σ S₂) :=
+add_monoid_algebra.is_scalar_tower
+instance [semiring R] [semiring S₁][comm_semiring S₂]
+  [semimodule R S₂] [semimodule S₁ S₂] [smul_comm_class R S₁ S₂] :
+  smul_comm_class R S₁ (mv_polynomial σ S₂) :=
+add_monoid_algebra.smul_comm_class
```

Comment:
These two shouldn't require commutativity, and should be able to stay at the top of the file; but perhaps you're just leaving this till later.